### PR TITLE
Documentation: Working With Openfire

### PR DESCRIPTION
**Pull request by Fishbowler**
Some useful tooling and links I liked. I was doing this anyway, and @guusdk said it could be a good contribution.

Fixed a few other bits I saw whilst I was in there.
Keen for feedback before it's "done". I'm not sure the tone is correct everywhere.

### CODE REVIEW

File: documentation/working-with-openfire.html [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    <p>There are plenty of extra parameters you can pass in for controlling which tests to run, who to authenticate as, and what the target of the tests should be. For example, this will run against xmpp.localhost.example instead of example.com, sets the auth credentials for the admin user, and runs only the MUC tests:</p>
+                    <p class="code">
+                        gradle integrationTest -Dsinttest.service=xmpp.localhost.example -Dsinttest.adminAccountUsername=admin -Dsinttest.adminAccountPassword=admin -Dsinttest.securityMode=disabled -Dsinttest.enabledTests=MultiUserChatIntegrationTest
+                    </p>
+                </div>
+
+                <h3>Dependencies</h3>
+                <p>Testing for out-of-date dependencies is important, especially as a release approaches. You can test for ones with known CVEs using dependency-check from within the project using what's built into the POM:</p>
+                <p class="code">
+                    mvn verify -P deps
```
Comment (GregDThomas) [inline documentation/working-with-openfire.html:89]: Also 
```
mvnw versions:display-dependency-updates
```
will show you all dependency updates, not just those with CVEs

**Comment by guusdk** [comment]
Thanks for doing this @Fishbowler - This will greatly help other developers!

For your viewing pleasure, you can look at HTML pages rendered, here: https://htmlpreview.github.io/?https://github.com/igniterealtime/Openfire/blob/36737d6978770d8c25e8f81b6160e52575faa18b/documentation/working-with-openfire.html

Things worth considering:

- Maybe make it a tad less informal: specifically the introduction. Explicitly claiming that this is _not_ official documentation takes away from the value of the text. Can we instead present this as "tips and tricks for developers"?
- Add a reference to open_chat for people wanting to discuss things with other developers.
- Invite people to add their own tips and tricks, by linking to a 'change-me-on-github' type of link to this page.

**Comment by Fishbowler** [comment]
Thanks both! Changes pushed!

**Comment by guusdk** [comment]
Looks good to me. I'm happy for this to be merged - or do you want to keep it around for more feedback?

**Comment by Fishbowler** [comment]
With the "edit on Github" link in there, I think we've got a good avenue for more feedback :)
Merge away!

**Pull request merged**